packages/cyberstorm/src/components/Switch.tsx

```diff
@@ -0,0 +1,32 @@
+import React, { PropsWithChildren, useCallback } from "react";
+import styles from "./componentStyles/Switch.module.css";
+import * as RadixSwitch from "@radix-ui/react-switch";
+
+export interface SwitchProps {
+  state: boolean;
+  switchCallback?: (state: boolean) => void;
+  disabled?: boolean;
+}
+
+export const Switch: React.FC<PropsWithChildren<SwitchProps>> = ({
+  state,
+  switchCallback = undefined,
+  disabled = false,
+}) => {
+  const onCheckedChange = useCallback(() => {
```

Comment:
I'm not sure I see the point of using `useCallBack` here, since the memoization probably adds more overhead than we save with it?

Reply:
IIRC there is an performance benefit, but then again I'm not 100% sure how the `useCallback` works. I've now left it out, in the name of simplifying code and me not knowing for sure.

Reply:
Recommended reading: https://kentcdodds.com/blog/usememo-and-usecallback